firmware/console/binary/tunerstudio_io_serial_ports.cpp

```diff
@@ -11,16 +11,23 @@
 #include "tunerstudio_io.h"
 #include "connector_uart_dma.h"
 
+// These may not be defined due to the HAL, but they're necessary for the compiler to do it's magic
+class UARTDriver;
```

Comment:
some of these are `struct`

Reply:
I swear I've seen that matter before, but it did compile, so ... ?  My google-fu is failing me on this one (trying to figure out when it's ok and when it's not).

Reply:
I think it chucks a warning for it (or maybe only clang does?) but it still works fine.